types/mongoose/index.d.ts

```diff
@@ -1545,6 +1545,9 @@ declare module "mongoose" {
       callback?: (err: any, doc: DocType | null, res: any) => void): DocumentQuery<DocType | null, DocType>;
     findOneAndUpdate(query: any | Query<any>, update: any,
       callback?: (err: any, doc: DocType | null, res: any) => void): DocumentQuery<DocType | null, DocType>;
+    findOneAndUpdate(query: any | Query<any>, update: any,
+      options: { upsert: true, new: true } & QueryFindOneAndUpdateOptions,
```

Comment:
QueryFindOneAndUpdateOptions already has `upsert` and `new` - it'd be a good idea to not override anything. Either `Pick` all the properties that aren't `new` or `upsert`, or use an [Omit type](http://ideasintosoftware.com/typescript-advanced-tricks/)

Reply:
I don't see how using Omit is any better...it's doing an "override" behind the scenes too and is just more complex overall. Intersecting the `{ upsert: boolean, new: boolean }` part of `QueryFindOneAndUpdateOptions` with  `{ upsert: true, new: true }` to narrow to true seems like the clearest approach to me.

Reply:
It's just that logically this  seems like `{upsert: true | undefined | boolean, new: true | undefined | boolean}`. If, however, this transpiles as expected, i.e. results in `DocType` and not `DocType | null` feel free to keep it.

Reply:
> It's just that logically this seems like `{upsert: true | undefined | boolean, new: true | undefined | boolean}`

That's what it would result in if it were a union, but as an intersection, it forces the key value to true.

> If, however, this transpiles as expected, i.e. results in DocType and not DocType | null feel free to keep it.

Cool, and yup, it does typecheck correctly.